### DISCUSSION TITLE
Change lazy_vars.CLIPBOARD

### DISCUSSION THREAD

**Discussion by voyeg3r** [L3MON4D3/LuaSnip#478] [Q&A] [answered]
In the file: `~/.local/share/nvim/site/pack/packer/start/LuaSnip/lua/luasnip/util/environ.lua` whe have:

```lua
function lazy_vars.CLIPBOARD() -- The contents of your clipboard
	return vim.fn.getreg('"', 1, true)
end
```

I would like to set the register to `+`, how can I achieve that?

**Comment by L3MON4D3** [comment, marked as answer]
Oh, this will be doable via custom env-variables, see #476.
So basically you can
- use a regular `functionNode` that returns `vim.fn.getreg("+", 1, true)` (not useable in lsp-snippets, but otherwise I'd just pick this)
- modify the source (eeehh)
- (wait until #476 is implemented, and use that :D)

**Reply by voyeg3r** [reply]
Thanks! I have changed the source for now. But I think the "CLIPBOARD" + makes more sense, isn't it true?

**Reply by L3MON4D3** [reply]
You're right, vscode's `$CLIPBOARD` (relevant because the variables are supposed to emulate vscodes behaviour) inserts `"+`, not `""` (your reason also stands of course, `:h quoteplus` explicitly associates `"+` with X11's CLIPBOARD).
I think we'll first do the customizable variables, then add a "source"/"provider" for registers, and then switch `$CLIPBOARDS` behaviour, with a notice to the new register-source for inserting the content of the unnamed register.

**Reply by leiserfg** [reply]
I just came to say
https://github.com/L3MON4D3/LuaSnip/pull/479
